### PR TITLE
Update to Quarkus 1.13

### PR DESCRIPTION
**Pull request by heubeck**
Update to Quarkus 1.13 and Kotlin 1.4.32.

The majority of changes is related to the new packaging format "fast-jar", others commented inline.

# Checklist

- [x] `mvn clean install` build and test completes

@karianna
@johnoliver

### CODE REVIEW

File: adoptopenjdk-api-v3-frontend/pom.xml [hunk 1]
```diff
@@ -151,29 +151,6 @@
         <sourceDirectory>src/main/kotlin</sourceDirectory>
         <testSourceDirectory>src/test/kotlin</testSourceDirectory>
         <plugins>
-            <plugin>
```
Comment (heubeck) [inline adoptopenjdk-api-v3-frontend/pom.xml:154]: The default artifact contains now all the compiled classes and resources, that's why attach-artifact shouldn't be required anymore and the jar-plugin is in again.

File: adoptopenjdk-api-v3-frontend/pom.xml [hunk 2]
```diff
@@ -279,21 +246,6 @@
             </activation>
             <build>
                 <plugins>
-                    <plugin>
```
Comment (heubeck) [inline adoptopenjdk-api-v3-frontend/pom.xml:282]: Reworked to current default native build configuration, but... native build isn't working anyhow, should it?

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/JsonSerializerConfig.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package net.adoptopenjdk.api.v3
+
+import io.quarkus.jsonb.JsonbConfigCustomizer
+import javax.inject.Singleton
+import javax.json.bind.JsonbConfig
+
+@Singleton
+class JsonSerializerConfig : JsonbConfigCustomizer {
```
Comment (heubeck) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/JsonSerializerConfig.kt:8]: `ContextResolver` `@Provider` wasn't used anymore, updated to recommended configuration.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt [hunk 4]
```diff
@@ -46,27 +47,31 @@ problems as an issue in the <a href=\"https://github.com/AdoptOpenJDK/openjdk-ap
...
         val ENABLE_PERIODIC_UPDATES: String = "enablePeriodicUpdates"
     }
 
-    private val apiDataStore: APIDataStore
+    private lateinit var apiDataStore: APIDataStore
     private val resourceClasses: Set<Class<out Any>>
     private val cors: Set<Any>
 
     @Inject
-    constructor(apiDataStore: APIDataStore) {
```
Comment (heubeck) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt:60]: constructor injection didn't work anymore, too.
jax-rs seems also not to require it, changed to setter injection and `@PostConstruct` initialization.

File: adoptopenjdk-api-v3-frontend/pom.xml (hunk 2 above)
Comment (johnoliver) [reply adoptopenjdk-api-v3-frontend/pom.xml:282]: As far as I know, nobody is using it, so I am fine with removing support until someone asks for it.

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+version: 2
```
Comment (johnoliver) [inline .github/dependabot.yml:1]: Whats the effect of this?

Comment (heubeck) [reply .github/dependabot.yml:1]: You mean the config-file or [dependabot](https://github.com/dependabot) itself?
Some time ago, dependabot was an externally managed bot, that creates PRs for updating the (maven/docker/gh-action/whatever) dependencies of repositories.
Meanwhile, dependabot is a Github build-in feature that can be enabled by this configuration.
Every day it checks the listed dependencies and creates PRs for each update available.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt (hunk 4 above)
Comment (johnoliver) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt:60]: I think this is probably a bug in ResteasyServerCommonProcessor, it seemed for me that even the presence of an empty constructor on the Application fixed this, i.e adding:

```
constructor() {}
```

To V3, and it seemed to work correctly. The other constructor with Inject was actually used.

I am a bit 50/50 on this as the empty constructor still means we need to have apiDataStore as lateinit, but I think I would prefer the constructor injection.

**Comment by johnoliver** [comment]
@heubeck One issue regarding the constructor Injection. Can see what you think on that one.

Other than that looks good and will merge depending on what you think on that issue.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt (hunk 4 above)
Comment (heubeck) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt:60]: Thank you for your thoughts, in general I also prefer constructor injection.
I did found discussions about this in Quarkus issues as well as on stackoverflow, the essence was, that one cannot relay that constructor injection works in the jax-rs Application class.

I try, if the constructor injection works if there's an additional empty constructor - if so, that's the winner.

**Comment by heubeck** [comment]
How about not having a field for the `apiDataStore` in class `V3` at all @johnoliver - please find the proposal updated.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt [hunk 6]
```diff
@@ -81,7 +76,11 @@ class V3 : Application {
         )
     }
 
-    private fun schedulePeriodicUpdates() {
+    /**
+     * Used to initialize the perioidc update scheduler of [APIDataStore]
+     */
```
Comment (johnoliver) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt:81]: looks good, but I would probably rename it from setApiDataStore as its not really a setter, this works for me:

```
    @PostConstruct
    @Inject
    fun schedulePeriodicUpdates(apiDataStore: APIDataStore) {
```

Comment (heubeck) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt:82]: Thank you @johnoliver, did not know, that `@PostConstruct` can take arguments.
Really like this approach, PR carries it now.

**Comment by johnoliver** [comment]
lgtm, thanks for the contribution, approved

**Review by johnoliver — APPROVED** [review]

**Pull request merged**